pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -1256,8 +1256,9 @@ private long validateSegmentStateForTable(String offlineTableName)
       TableSegmentValidationInfo tableSegmentValidationInfo =
           JsonUtils.stringToObject(response, TableSegmentValidationInfo.class);
       if (!tableSegmentValidationInfo.isValid()) {
-        throw new ControllerApplicationException(LOGGER, "Table segment validation failed",
-            Response.Status.PRECONDITION_FAILED);
+        String error = String.format("Table segment validation failed. error=%s",
```

Comment:
(minor) We have started an effort to avoid using `String.format()` and replace it with regular concatenation. Same for other places

Reply:
what's the reasoning?

Reply:
More context can be found in #14404